File: app/src/main/java/com/miichisoft/plesuxtranining/adapter/ListAdapter.java
package com.miichisoft.plesuxtranining.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.miichisoft.plesuxtranining.R;
import com.miichisoft.plesuxtranining.model.ItemAdapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class ListAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    private List<ItemAdapter> mList;

    public ListAdapter(List<ItemAdapter> list) {
        mList = list;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        ItemAdapter itemAdapter = mList.get(position);
        ((ViewHolder) holder).mTv_name.setText(itemAdapter.getText());
        ((ViewHolder) holder).mImg.setImageResource(itemAdapter.getImage());
    }

    @Override
    public int getItemCount() {
        return mList.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        public TextView mTv_name;
        public ImageView mImg;

        public ViewHolder(View itemView) {

            super(itemView);
            mTv_name = (TextView) itemView.findViewById(R.id.tv_name);
            mImg = (ImageView) itemView.findViewById(R.id.img_item);

        }
    }
}

File: app/src/main/java/com/miichisoft/plesuxtranining/adapter/ListAdapter.kt
package com.miichisoft.plesuxtranining.adapter

import android.content.Intent
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.miichisoft.plesuxtranining.FruitDetailsActivity
import com.miichisoft.plesuxtranining.R
import com.miichisoft.plesuxtranining.RecyclerViewActivity
import com.miichisoft.plesuxtranining.model.ItemAdapter

class ListAdapter(private val mList: List<ItemAdapter>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.list_item, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val itemAdapter = mList[position]
        (holder as ViewHolder).mTv_name.text = itemAdapter.text
        holder.mImg.setImageResource(itemAdapter.image)
        holder.itemView.setOnClickListener { v ->
            Log.d("Clicked", "Clicked")
            val intent: Intent = Intent(v.context, FruitDetailsActivity::class.java)
            intent.putExtra("fName", holder.mTv_name.text)
            v.context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return mList.size
    }

    internal class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mTv_name: TextView = itemView.findViewById<View>(R.id.tv_name) as TextView
        var mImg: ImageView = itemView.findViewById<View>(R.id.img_item) as ImageView
    }
}